pwiz_tools/Skyline/Controls/Graphs/DetectionsHistogramPane.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.Remoting.Messaging;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using pwiz.Common.Collections;
+using pwiz.Skyline.Controls.SeqNode;
+using pwiz.Skyline.Model;
+using pwiz.Skyline.Properties;
+using ZedGraph;
+using Settings = pwiz.Skyline.Controls.Graphs.DetectionsGraphController.Settings;
+
+namespace pwiz.Skyline.Controls.Graphs
+{
+    public class DetectionsHistogramPane : DetectionsPlotPane
+    {
+
+        public DetectionsHistogramPane(GraphSummary graphSummary) : base(graphSummary )
+        {
+            XAxis.Type = AxisType.Ordinal;
+            XAxis.Title.Text = Resources.DetectionHistogramPane_XAxis_Name;
+        }
+
+        public override ImmutableList<int> GetDataSeries()
+        {
+            return TargetData.Histogram;
+        }
+
+
+        public override void UpdateGraph(bool selectionChanged)
+        {
+            if (!_detectionData.IsValid)

```

Comment:
BUG: I believe this check needs to go after the line where you assign _detectionData to the value from the cache.

Otherwise, if you get yourself in a state where the detectionData is invalid, it will never fetch new data, and the graph will never update.

Reply:
Thank you, Nick, appreciate your comments. Good catch on _detectionData.IsValid logic, this definitely should be fixed. I didn't see the other exception you mentioned, but I agree that Dispose in not really necessary there.